### ISSUE TITLE
Scan Report Mapping Rules Modals

### ISSUE THREAD

**Issue by AndyRae** [Health-Informatics-UoN/Carrot-Mapper#751]
# Summary

Rewrite the Scan Reporting Mapping Rules modals in Nextjs, depends on #750 

# Tasks
- [ ] Add "Show Summary View" modal
     - [ ] Add "simple" table rendering the summary data
     - [ ] This data is filtered on `source_concept_id` and only includes Rules `destination_field.name` that contain it. 
- [ ] Add "Analyse Rules" modal 
     - [ ] Find an example of "Analyse Rules" being populated (in dev, test, or production)
     - [ ] Mirror the existing view (this is unknown until we see one) 
     - [ ] Add "simple" table representing the Rules Analysis

# Acceptance Criteria

- [ ] Working modals to represent the Scan Report Mapping Rules

**Comment by AndrewThien** [comment]
Need more information on Analyse Rules modal

**Comment by spco** [comment]
What information do you need?

**Comment by AndrewThien** [comment]
Hi Sam, 
Thanks for asking.
Emeka @Dev-Akashili  has been working on this, but I also have some questions:
I knew the purpose of this function but I didn't know and couldn't see how its result looked like on the web app. This is because the API endpoint /api/analyse/<SR-ID> wasn't returning anything and has been broken (after a while of running). So, the question is: Do we need to re-build the API and how can we get more information about the ancestors or descendants of a scan report (how they looked like and how they need to be represented on the UI, ...)  

![Image](https://github.com/user-attachments/assets/14ca3973-91ab-441d-9335-be110710974d)



![Image](https://github.com/user-attachments/assets/4da4d1bd-678a-44e2-bb95-a15df953a3f4)

**Comment by spco** [comment]
No problem! See https://github.com/Health-Informatics-UoN/Carrot-Mapper/pull/393 for an example of how it looks when populated.

The idea of this endpoint is to compare all Concepts used in this Scan Report, and compare these to all Concepts in _all other_ Scan Reports. OMOP Concepts exists in a non-strict hierarchy. Each Concept can have multiple parents and multiple children. Parents are very generic e.g. "heart condition", children are more specific e.g. "atrial fibrillation of the blah blah blah".

These relationships are defined as ancestors and descendants. See the `omop.concept_ancestor` table, which also then shows how far apart 2 concepts are in the hierarchy (`min_levels_of_separation`). The `omop.concept_relationship` table shows all pairs of concepts and their relationship _type_. We're interested in the case where that type is 

We then display any concepts present in this Scan Report, that have ancestor or descendant Concepts in any other Scan Report. In this way, the user can see where the Concepts they've chosen are "misaligned" with Concepts used elsewhere. It's up to the user to decide whether they take any action based on this or not.

**Comment by spco** [comment]
As for the endpoint, I've no idea what state it is in, so it may well need some work to bring it back online.

And a note - clearly this is an incredibly expensive operation if you have a lot of Concepts in the current SR and lots in other SRs. Running this on-the-fly is probably not reasonable, and we need to think about caching/offlining this calculation.

**Comment by AndrewThien** [comment]
Thanks a lot for the valuable information and advice, Sam. 
Seems to be a lot of work here, tbh.
We will discuss more about this soon and probably when @AndyRae comes back.

**Comment by AndyRae** [comment]
> As for the endpoint, I've no idea what state it is in, so it may well need some work to bring it back online.
> 
> And a note - clearly this is an incredibly expensive operation if you have a lot of Concepts in the current SR and lots in other SRs. Running this on-the-fly is probably not reasonable, and we need to think about caching/offlining this calculation.

So the endpoint is still working as intended. But as Carrot now deals with scale better in other areas... the analyse function can't deal with that. The existing button the React app is actually now a quick way for any user to bring the app to a halt (unbeknownst to them), which is why it might appear to not be working at first.

So we will not reimplement the existing analyse rules modal - as we need to remove these kind of liabilities. We will remove the button for now, and plan to move this to a serverless function in future work. This lends to our work with "job" status for a Scan Report also, as in a user clicks the analyse button - and can be notified when it is completed.

**Comment by spco** [comment]
Sounds a good plan. We have made a bit of a song and dance in the paper about the fact that we do highlight misaligned mappings, so we should definitely reinstate this soon. But for now it's not working anyway (besides the fact it crashes as Andy says, the shonky pagination means it also only ever displays info about the currently viewed page, so about 4 or 5 ScanReportConcepts at any one time).
